src/verify.c

```diff
@@ -1153,6 +1180,30 @@ enum swupd_code verify_main(void)
 	 * naming convention: All exit goto labels must follow the "brick_the_system_and_FOO:" pattern
 	 */
 
+	/* track bundles specified by user */
+	if (cmdline_option_install && cmdline_option_bundles) {
+		/* this is a fresh install so the tracking directory
+		 * needs to be created */
+		char *new_os_statedir = mk_full_filename(globals.path_prefix, "/var/lib/swupd");
+		bool tracking = init_tracking_dir(new_os_statedir, ".init");
+		if (tracking) {
+			for (iter = cmdline_option_bundles; iter; iter = iter->next) {
+				char *bundle = iter->data;
+				/* make sure the bundle was in fact valid and will
+				 * be installed before creating the tracking file */
+				if (list_search(bundles_subs, bundle, subscription_bundlename_strcmp)) {
```

Comment:
Is this list_search comparision checking if the bundle was installed manually by user or included as a includes?

If it added manually, then we create a tracking file for it?

Reply:
yes, the bundles the user is specifically attempting to install are in the  cmdline_option_bundles list, however there is a chance the user specified an invalid bundle, and we don't want to create a tracking file for an invalid bundle. So by making sure there is a subscription that matches each bundle in the list we make sure the bundle to track was in fact valid.